web/src/lib/components/shared-components/map/map.svelte

```diff
@@ -50,21 +43,48 @@
     }
   }
 
-  export let onOpenInMapView: (() => Promise<void> | void) | undefined = undefined;
-  export let onSelect: (assetIds: string[]) => void = () => {};
-  export let onClickPoint: ({ lat, lng }: { lat: number; lng: number }) => void = () => {};
+  interface Props {
+    mapMarkers: MapMarkerResponseDto[];
+    showSettingsModal?: boolean | undefined;
+    zoom?: number | undefined;
+    center?: LngLatLike | undefined;
+    hash?: boolean;
+    simplified?: boolean;
+    clickable?: boolean;
+    useLocationPin?: boolean;
+    onOpenInMapView?: (() => Promise<void> | void) | undefined;
+    onSelect?: (assetIds: string[]) => void;
+    onClickPoint?: ({ lat, lng }: { lat: number; lng: number }) => void;
+    popup?: import('svelte').Snippet<[{ marker: MapMarkerResponseDto }]>;
+  }
+
+  let {
+    mapMarkers = $bindable(),
+    showSettingsModal = $bindable(undefined),
+    zoom = undefined,
+    center = $bindable(undefined),
+    hash = false,
+    simplified = false,
+    clickable = false,
+    useLocationPin = false,
+    onOpenInMapView = undefined,
+    onSelect = () => {},
+    onClickPoint = () => {},
+    popup,
+  }: Props = $props();
 
-  let map: maplibregl.Map;
+  let map: maplibregl.Map | undefined = $state();
   let marker: maplibregl.Marker | null = null;
 
-  // svelte-ignore reactive_declaration_non_reactive_property
-  $: style = (async () => {
-    const config = await getServerConfig();
-    const theme = $mapSettings.allowDarkMode ? $colorTheme.value : Theme.LIGHT;
-    const styleUrl = theme === Theme.DARK ? config.mapDarkStyleUrl : config.mapLightStyleUrl;
-    const style = await fetch(styleUrl).then((response) => response.json());
-    return style as StyleSpecification;
-  })();
+  let style = $derived(
+    (async () => {
+      const config = await getServerConfig();
+      const theme = $mapSettings.allowDarkMode ? $colorTheme.value : Theme.LIGHT;
+      const styleUrl = theme === Theme.DARK ? config.mapDarkStyleUrl : config.mapLightStyleUrl;
+      const style = await fetch(styleUrl).then((response) => response.json());
+      return style as StyleSpecification;
+    })(),
+  );
```

Comment:
```suggestion
  const theme = $derived($mapSettings.allowDarkMode ? $colorTheme.value : Theme.LIGHT);
  const styleUrl = $derived(theme === Theme.DARK ? $serverConfig.mapDarkStyleUrl : $serverConfig.mapLightStyleUrl);
  const style = $derived(fetch(styleUrl).then((response) => response.json()));
```

Fix for the following scenario:
- Enable dark mode
- Disable `Allow dark mode` in map settings
- Save